app/main.py

```diff
@@ -1,3 +1,21 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
+
+    memory = {}
+
+    @functools.wraps(func)
+    def wrapper(*args):
+        nonlocal memory
```

Comment:
You don't need to use nonlocal here. It can be used for reassigning variables. If you need just to change some fields of dict it is unnecessary.